src/Microsoft.AspNetCore.Mvc.Razor/Internal/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -56,7 +57,6 @@ public IEnumerable<TagHelperDescriptor> CreateDescriptors(string assemblyName)
                     StringComparison.Ordinal));
 
             var tagHelperDescriptors = viewComponentDescriptors
-                .Where(descriptor => !descriptor.Parameters.Any(parameter => parameter.ParameterType.GetTypeInfo().IsGenericType))
```

Comment:
I tested this and you are right. Checking for generic params in MethodInfo is the right way to go. I'll send a PR.